modules/bday.js

```diff
@@ -0,0 +1,227 @@
+const axios = require('axios')
+const { WebClient } = require('@slack/web-api')
+
+
+const { SLACK_OAUTH } = process.env
+const web = new WebClient(SLACK_OAUTH)
+
+const invalidInputNotice = (response_url, command) => (
+  axios.post(response_url, {
+    response_type: 'ephemeral',
+    blocks: [
+      { type: 'divider' },
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `Failed to proccess command */bday ${command}*. _Hint: be sure to link to a valid URL/valid user!_`,
+        },
+      },
+    ],
+  })
+)
+const confirmChannel = (response_url, cmd, channel) => (
+  axios.post(response_url, {
+    response_type: 'ephemeral',
+    blocks: [
+      { type: 'divider' },
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `Please run the */bday ${cmd}* command in channel: #${channel}.`,
+        },
+      },
+    ],
+  })
+)
+
+const worker = async ({ command, channel_id, channel_name, response_url }) => {
+  // send card for members to sign
+  if (command.includes('sign') && command.includes('for')) {
+    // make sure to execute command from #general
+    if (channel_name !== 'general') {
+      return confirmChannel(response_url, 'sign', 'general')
+    }
+    // sign URL for @user_id|user_name
+    const R = /<(?<MIRO_URL>.*)> for <@(?<BDAY_USER_ID>.*)\|(?<BDAY_USER_NAME>.*)>/
+    const matches = command.match(R)
+    // notify user of invalid input
+    if (!matches) {
+      return invalidInputNotice(response_url, command)
+    }
+    // send to every member in #general excluding bday person
+    const { groups: { MIRO_URL, BDAY_USER_ID, BDAY_USER_NAME } } = matches
+    const { members } = await web.conversations.members({ channel: channel_id })
+    members.splice(members.findIndex((m) => m === BDAY_USER_ID), 1)
+    return Promise.all(
+      members.map(async (channel) => await web.chat.postMessage({
+        channel,
+        blocks: [
+          { type: 'divider' },
+          {
+            type: 'section',
+            text: {
+              type: 'mrkdwn',
+              text: [
+                `:tada: *${BDAY_USER_NAME}'s* birthday is coming up soon! :birthday:`,
+                `:tada: When you get a chance, visit ${MIRO_URL} to sign their birthday card!`,
+              ].join('\n'),
+            },
+          },
+        ],
+      }))
+    )
```

Comment:
`async/await` here is redundant as `web.chat.postMessage` already returns a promise.

more importantly, we should use https://api.slack.com/methods/conversations.open which natively supports a list of `users` to send to, which is much more efficient than async invocation of N API calls which is prone to be rate limited (imagine the workplace grows into hundreds of coworkers)

Reply:
tried `conversations.open`, it opens up a group chat with all members in the `users` array when there's more than 1 user, we'd prefer having the bot sending dm to each member lol (we'll take a close look, e.g. how apps like `donut` sends their app messages in bulk within rate limit)